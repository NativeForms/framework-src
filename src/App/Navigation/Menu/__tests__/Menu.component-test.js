import React from 'react';
import { ListItem } from 'native-base';
import sinon from 'sinon';
import { shallow, mount } from 'enzyme';

// app
import MenuComponent from '../Menu.component';

// framework
import routes from '../../../../Framework';

const props = {
  navigation: { navigate: sinon.spy() }
};
const routeNames = routes.map(route => route.routeName);

test('MenuComponent should render correctly', () => {
  const wrapper = shallow(<MenuComponent {...props} />);
  expect(wrapper).toMatchSnapshot();
});

['Home', ...routeNames].forEach(route => {
  describe(`component <ListItem>${route}</ListItem>`, () => {
    const consoleError = global.console.error;
    let menuProps;
    beforeEach(() => {
      global.console.error = jest.fn();
      menuProps = {
        navigation: { navigate: sinon.spy() }
      };
    });

    it(`should navigate to ${route} on Item press`, () => {
      const wrapper = mount(<MenuComponent {...menuProps} />);
      wrapper.find(ListItem).findWhere(node => node.text() === route).at(1).props()
      .onPress();
      expect(menuProps.navigation.navigate.calledOnce).toBeTruthy();
    });

    afterEach(() => {
      global.console.error = consoleError;
    });
  });
});
