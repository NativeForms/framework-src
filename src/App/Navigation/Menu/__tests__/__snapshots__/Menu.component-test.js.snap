// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuComponent should render correctly 1`] = `
<Styled(Content)>
  <Styled(List)>
    <Styled(ListItem)
      button={true}
      onPress={[Function]}
    >
      <Styled(Text)>
        Home
      </Styled(Text)>
    </Styled(ListItem)>
    <Styled(ListItem)
      itemDivider={true}
    >
      <Styled(Text)>
        Items
      </Styled(Text)>
    </Styled(ListItem)>
    <Styled(ListItem)>
      <Styled(Text)>
        Item 1
      </Styled(Text)>
    </Styled(ListItem)>
    <Styled(ListItem)>
      <Styled(Text)>
        Item 2
      </Styled(Text)>
    </Styled(ListItem)>
    <Styled(ListItem)>
      <Styled(Text)>
        Item 3
      </Styled(Text)>
    </Styled(ListItem)>
  </Styled(List)>
  <Styled(ListItem)
    itemDivider={true}
  >
    <Styled(Text)>
      Demos
    </Styled(Text)>
  </Styled(ListItem)>
  <Styled(ListItem)
    button={true}
    onPress={[Function]}
  >
    <Styled(Text)>
      Counter
    </Styled(Text)>
  </Styled(ListItem)>
</Styled(Content)>
`;
